Sprint_5 develop

blinkovv opened the pull request.

blinkovv committed:
Добавил .gitignore

blinkovv committed:
Добавил директорию и файлы в нее configuration.py, locators.py and test_successful_registration_and_login

blinkovv committed:
Изменил содержимое README.md

AlexKitIt commented on tests/configuration.py:
```diff
@@ -0,0 +1,30 @@
+import random
+
+
+EMAIL = "GrantBlinkov11776@yandex.ru"
+PASSWORD = "Qwe123Asd456!"
+URL = "https://stellarburgers.nomoreparties.site/"
+N = 13
```
Можно лучше: тестовые данные стоит хранить отдельно в data модуле, так поддерживать и переиспользовать будет проще

AlexKitIt commented on tests/locators.py:
```diff
@@ -0,0 +1,27 @@
+input_name = '//*[@id="root"]/div/main/div/form/fieldset[1]/div/div/input'  # Регистрация. Поле Имя
```
Нужно исправить: здесь и далее: использование порядкового номера элемента в dom является весьма нестабильным способом поиска, необходимо привязываться к атрибутам конкретного элемента
Кроме того ходить от root и использовать абсолютные пути нельзя - очень ненадежно, добавится на пути новый элемент - тест ляжет
Все локаторы необходимо переписать

AlexKitIt commented on tests/test_successful_registration_and_login.py:
```diff
@@ -0,0 +1,100 @@
+import configuration
+
+
+def test_invalid_password(driver):   # Невалидный пароль, ошибка
```
Нужно исправить: здесь и далее по тестам: тестовые методы в модулях объединяются в тестовые классы

AlexKitIt commented on tests/test_successful_registration_and_login.py:
Нужно исправить: тут должно быть несоклько тестовых модулей с тестами, объединенными по общей функциональности

AlexKitIt replied:
Повторный коммент: не один модуль с кучей классов а несколько!

AlexKitIt commented on tests/test_successful_registration_and_login.py:
```diff
@@ -0,0 +1,100 @@
+
+
+def test_invalid_password(driver):   # Невалидный пароль, ошибка
+    driver.get("https://stellarburgers.nomoreparties.site/register")
```
Это вынесено в config, почему тут не используется?

AlexKitIt commented on tests/test_successful_registration_and_login.py:
```diff
@@ -0,0 +1,100 @@
+    driver.find_element(By.XPATH, locators.button_constructor).click()  # Переход из ЛК в Конструктор
+    driver.find_element(By.XPATH, locators.button_personal_account).click()  # Переход в ЛК
+    driver.find_element(By.CSS_SELECTOR, locators.button_logotype).click()  # Переход по клику на логотип Stellar Burgers.
+    assert 'Соберите бургер' in driver.find_element(By.XPATH, '//*[@id="root"]/div/main/section[1]/h1').text
```
Нужно исправить: все локаторы хранятся отдельно от тестов в модуле locators

AlexKitIt commented on tests/test_successful_registration_and_login.py:
```diff
@@ -0,0 +1,100 @@
+    assert "Вход" in successful_registrations
+
+
+def test_constructor_navigation(driver):
```
Нужно исправить: тестов на переход по табам должно быть три отдельных, по одному на каждый таб

blinkovv committed:
Добавлены константы registration and login (URL)

blinkovv committed:
Тестовые методы объединены в тестовые классы
Создано несколько тестовых модулей с тестами
Тесты на переход по табам разделены на три отдельных теста

blinkovv committed:
Переписал все локаторы, уйдя от root

AlexKitIt commented on tests/configuration.py:
```diff
@@ -0,0 +1,32 @@
+res = ''.join(random.choices(string.ascii_letters, k=N))
+
+
+def generate_email():
+    email = ["@yandex.ru", "@mail.ru", "@gmail.com"]
+    rand_idx = random.randrange(len(email))
+    random_num = email[rand_idx]
+    res = ''.join(random.choices(string.ascii_letters, k=13))
+    return f'{res}{random_num}'
+
+
+def generate_name():
+    res = ''.join(random.choices(string.ascii_letters, k=13))
+    return res
+
+
+def generate_password():
+    res = ''.join(random.choices(string.ascii_letters, k=13))
+    return f'{res}{random.randint(0, 9)}'
```
Можно лучше: статичные данные стоит отделить в data модули  а методы в helpers
